DSpace/dspace-angular
# Made it possible to reorder the login methods

## References
* Fixes #2365
* Maintenance PR #2488

## Description
Made it possible to reorder the login methods based on the order provided in the config
<img width="290" alt="image" src="https://github.com/DSpace/dspace-angular/assets/43750381/eb2a978f-f4ed-4f73-b24f-8ba2a221fc9a">

## Instructions for Reviewers
List of changes in this PR:
* Save the display position in the `AuthMethod`
* Moved the user registration & password reset link to the `LogInPasswordComponent`
* Refactored the way `authMethods` are retrieved, now the subscription is automatically taken care of by the `async` pipe

**Guidance for how to test/review this PR:**
- Set the following config in the backend:
   ```cfg
   plugin.sequence.org.dspace.authenticate.AuthenticationMethod = org.dspace.authenticate.OrcidAuthentication, org.dspace.authenticate.ShibAuthentication, org.dspace.authenticate.PasswordAuthentication, org.dspace.authenticate.OidcAuthentication
   ```
- Check that the authentication methods are displayed in the correct order

## Checklist
